src/test/java/com/iexec/blockchain/tool/ChainConfigTest.java

```diff
@@ -0,0 +1,129 @@
+package com.iexec.blockchain.tool;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class ChainConfigTest {
+    private ChainConfig chainConfig;
+
+    @BeforeEach
+    void setUp() {
+        this.chainConfig = new ChainConfig(
+                Validation.buildDefaultValidatorFactory().getValidator()
+        );
+    }
+
+    static Stream<Arguments> validData() {
+        return Stream.of(
+                Arguments.of(100, "http://localhost:8545", "0xBF6B2B07e47326B7c8bfCb4A5460bef9f0Fd2002", 1),
+                Arguments.of(42, "https://localhost:8545", "0x0000000000000000000000000000000000000000", 10),
+                Arguments.of(10, "https://www.classic-url.com", "0xBF6B2B07e47326B7c8bfCb4A5460bef9f0Fd2002", 42),
+                Arguments.of(1, "http://ibaa.iex.ec:443/test?validation=should:be@OK", "0xBF6B2B07e47326B7c8bfCb4A5460bef9f0Fd2002", 100)
+        );
+    }
+
+    @ParameterizedTest
```

Comment:
Playing with reflection on the same instance for different tests seems a little dangerous for me. I find this more readable and less complicated:
1. Add `@AllArgsConstructor` to ChainConfig.
2. 
```
new ChainConfig(100, "http://localhost:8545", "0xBF6B2B07...", 1, validator).validate()
new ChainConfig(100, "https://www.classic-url.com", "0xBF6...", 1, validator).validate()
...
```
What do you think?

Reply:
Ok for the first suggestion. The second would not work and be clean because:
1. `validate` is still private (we may set it package-private, ok.);
2. with `@AllArgsConstructor`, we'll also have to explicitly set the other values (`isSidechain`, `gasPriceMultiplier`, `gasPriceCap` and `brokerUrl`). I prefer to keep the tested values outside of the test logic, so the `@ParameterizedTest` seems to be the best option, there.

Reply:
Why don't we test fields independently? It avoids missing some conditions

Reply:
We could do so, but that would make a long list of input data. Not sure it would be more readable. But if you think it'd be better, I'll do it.